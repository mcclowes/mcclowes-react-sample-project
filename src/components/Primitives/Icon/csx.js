import styled from "styled-components/macro";

export const IconWrapper = styled.svg`
  fill: ${(props) => props.theme.colors.black};
  width: ${(props) => (props.width ? props.width : "16px")};
  height: ${(props) => (props.height ? props.height : "16px")};
`;

IconWrapper.defaultProps = {
  theme: {
    colors: {
      black: "#333",
    },
  },
};
