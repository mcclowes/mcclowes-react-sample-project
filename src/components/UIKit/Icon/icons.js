const icons = {
  CLOCK:
    "M16,2C8.27,2,2,8.27,2,16s6.27,14,14,14s14-6.27,14-14S23.73,2,16,2z M18.71,21.54l-4.24-4.24l0.01-0.01C14.18,16.93,14,16.49,14,16V6h4v9.17l3.54,3.54L18.71,21.54z",
  CLOSE:
    "M18.82,16l7.09,7.09l-2.82,2.82L16,18.82l-7.09,7.09l-2.82-2.82L13.18,16L6.09,8.91l2.82-2.82L16,13.18l7.09-7.09l2.82,2.82L18.82,16z",
  DELETE:
    "M6,10V28a2,2,0,0,0,2,2H24a2,2,0,0,0,2-2V10ZM26,4H22L20,2H12L10,4H6A2,2,0,0,0,4,6V8H28V6A2,2,0,0,0,26,4Z",
  DOWN: "M3.75,6.13,16,18.38,28.25,6.13,32,9.87l-16,16L0,9.87Z",
  EDIT:
    "M27.79,9.21,25,12,20,7l2.79-2.79a1,1,0,0,1,1.42,0l3.58,3.58A1,1,0,0,1,27.79,9.21ZM4,23v5H9L23.5,13.5l-5-5Z",
  ELLIPSIS:
    "M19,16c0,1.66-1.34,3-3,3s-3-1.34-3-3s1.34-3,3-3S19,14.34,19,16z M5,13c-1.66,0-3,1.34-3,3s1.34,3,3,3s3-1.34,3-3S6.66,13,5,13z M27,13c-1.66,0-3,1.34-3,3s1.34,3,3,3s3-1.34,3-3S28.66,13,27,13z",
  EXPORT: "M26,12H21V22H11V12H6L16,2Zm2,14H4v4H28Z",
  HELP:
    "M16,2C8.27,2,2,8.27,2,16s6.27,14,14,14s14-6.27,14-14S23.73,2,16,2z M16,28C9.37,28,4,22.63,4,16C4,9.37,9.37,4,16,4s12,5.37,12,12C28,22.63,22.63,28,16,28z M18,10c0,1.1-0.9,2-2,2s-2-0.9-2-2s0.9-2,2-2S18,8.9,18,10z M16,24L16,24c-1.1,0-2-0.9-2-2v-6c0-1.1,0.9-2,2-2h0c1.1,0,2,0.9,2,2v6C18,23.1,17.1,24,16,24z",
  OPEN:
    "M18,26v2a2,2,0,0,1-2,2H4a2,2,0,0,1-2-2V16a2,2,0,0,1,2-2H6V26ZM26,6H12V20H26V6m1.88-4A2.12,2.12,0,0,1,30,4.12V21.88A2.12,2.12,0,0,1,27.88,24H10.12A2.12,2.12,0,0,1,8,21.88V4.12A2.12,2.12,0,0,1,10.12,2Z",
  SETTINGS:
    "M29.26,19,26,16.77c0-.25,0-.51,0-.77s0-.51,0-.77L29.26,13a1,1,0,0,0,.36-1.36l-3-5.2A1,1,0,0,0,25.26,6l-3.6,1.72A10.54,10.54,0,0,0,20.32,7L20,3a1,1,0,0,0-1-1H13a1,1,0,0,0-1,1l-.31,4a9.42,9.42,0,0,0-1.35.77L6.74,6a1,1,0,0,0-1.36.36l-3,5.2A1,1,0,0,0,2.74,13L6,15.21c0,.26,0,.52,0,.79s0,.52,0,.78L2.74,19a1,1,0,0,0-.36,1.36l3,5.2A1,1,0,0,0,6.74,26l3.59-1.73A9.75,9.75,0,0,0,11.7,25l.3,4a1,1,0,0,0,1,1h6a1,1,0,0,0,1-1l.3-4a10.49,10.49,0,0,0,1.35-.77L25.26,26a1,1,0,0,0,1.36-.36l3-5.2A1,1,0,0,0,29.26,19ZM16,21a5,5,0,1,1,5-5A5,5,0,0,1,16,21Z",
  TICK:
    "M16.35,21.34L16.35,21.34l-2.83,2.83l-2.83-2.83l-6.36-6.36l2.83-2.83l6.36,6.36L24.84,7.2l2.83,2.83L16.35,21.34z",
};

export default icons;
