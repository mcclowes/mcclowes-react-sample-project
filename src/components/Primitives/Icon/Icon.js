import icons from "./icons";
import PropTypes from "prop-types";
import React from "react";
import { IconWrapper } from "./csx";

const Icon = (props) => {
  const { path, width, height, icon } = props;

  return (
    <IconWrapper
      viewBox="0 0 32 32"
      focusable="false"
      width={width}
      height={height}
    >
      {icon ? <path d={icons[icon]} /> : <path d={path} />}
    </IconWrapper>
  );
};

Icon.propTypes = {
  height: PropTypes.any,
  icon: PropTypes.string,
  path: PropTypes.string,
  width: PropTypes.any,
};

export default Icon;

export { icons };
