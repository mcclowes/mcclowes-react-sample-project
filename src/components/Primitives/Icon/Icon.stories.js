import React from "react";
import Icon, { icons } from "./Icon";

export default { title: "Icon" };

export const simple = () => {
  return <Icon icon="TICK" />;
};

export const withPath = () => {
  return <Icon path={icons.TICK} />;
};

export const all = () => {
  return (
    <>
      {Object.values(icons).map((icon) => (
        <Icon path={icon} />
      ))}
    </>
  );
};
