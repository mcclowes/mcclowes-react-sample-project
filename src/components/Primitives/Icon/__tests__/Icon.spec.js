import React from "react";
import Icon from "../Icon";
import icons from "../icons";
import { shallow } from "enzyme";

const defaultProps = {
  path: icons.TICK,
};

const renderComponent = (newProps) => {
  const props = {
    ...defaultProps,
    ...newProps,
  };

  return shallow(<Icon {...props} />);
};

let wrapper;

describe("Icon", () => {
  it("renders as expected", () => {
    wrapper = renderComponent();

    expect(wrapper).toMatchSnapshot();
  });
});
