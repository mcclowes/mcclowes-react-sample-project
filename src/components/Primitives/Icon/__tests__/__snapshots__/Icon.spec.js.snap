// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders as expected 1`] = `
<csx__IconWrapper
  focusable="false"
  theme={
    Object {
      "colors": Object {
        "black": "#333",
      },
    }
  }
  viewBox="0 0 32 32"
>
  <path
    d="M16.35,21.34L16.35,21.34l-2.83,2.83l-2.83-2.83l-6.36-6.36l2.83-2.83l6.36,6.36L24.84,7.2l2.83,2.83L16.35,21.34z"
  />
</csx__IconWrapper>
`;
